pyfr/backends/cuda/types.py

```diff
@@ -27,18 +27,17 @@ def onalloc(self, basedata, offset):
 
     def _get(self):
         # Allocate an empty buffer
-        buf = np.empty((self.nrow, self.leaddim), dtype=self.dtype)
+        buf = np.empty((self.nblocks, self.nrow, self.leaddim),
```

Comment:
Why is nblocks needed here?

Reply:
unpack expects a 3d array. I can replace with 1 if you think that it is better.

Reply:
Or you can pass as buf[None, :, :] which will slot in the extra dim.